File: Study Plans/Binary Search/Python/search-a-2d-matrix.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        
        for i in matrix:
            
            low = 0;
            high = len(i)-1
            
            while low <= high:
                
                mid = (low+high)//2
                
                if i[mid] == target:
                    return True
                
                if i[mid] < target:
                    low = mid + 1
                else:
                    high = mid - 1
        
        return False

File: Study Plans/Binary Search/Java/Search-a-2D-Matrix.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        
        for (int[] i:matrix) {
            
            int low = 0;
            int high = i.length-1;
            
            while (low<=high) {
                
                int mid = (low+high)/2;
                
                int res = i[mid];
                
                if (res == target) {
                    return true;
                }
                
                if (res < target) {
                    low = mid + 1;
                } else {
                    high = mid - 1;
                }
            }
        }
        return false;
    }
}
